### ISSUE TITLE
py-matplotlib build fails during `set_backend`

### ISSUE THREAD

**Issue by sethrj** [spack/spack#12230]
It looks like #12164 might have broken this? @adamjstewart 

### Steps to reproduce the issue

```console
$ spack install py-matplotlib
...
==> Installing py-matplotlib
==> Searching for binary cache of py-matplotlib
==> Warning: No Spack mirrors are currently configured
==> No binary for py-matplotlib found: installing from source
==> Fetching https://pypi.io/packages/source/m/matplotlib/matplotlib-3.1.1.tar.gz
######################################################################## 100.0%
==> Staging archive: /projects/spack2/var/spack/stage/py-matplotlib-3.1.1-6k3m7aerxlh3q5sikixccqle7or3x25o/matplotlib-3.1.1.tar.gz
==> Created stage in /projects/spack2/var/spack/stage/py-matplotlib-3.1.1-6k3m7aerxlh3q5sikixccqle7or3x25o
==> No patches needed for py-matplotlib
==> Building py-matplotlib [PythonPackage]
==> Executing phase: 'build'
==> Error: NameError: global name 'backend' is not defined

/projects/spack2/var/spack/repos/builtin/packages/py-matplotlib/package.py:145, in set_backend:
        142        with open('setup.cfg', 'w') as setup:
        143            # Default backend
        144            setup.write('[rc_options]\n')
  >>    145            setup.write('backend = ' + backend + '\n')

See build log for details:
  /projects/spack2/var/spack/stage/py-matplotlib-3.1.1-6k3m7aerxlh3q5sikixccqle7or3x25o/spack-build-out.txt
```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a 
Linux node1 3.10.0-327.10.1.el7.x86_64 #1 SMP Sat Jan 23 04:54:55 EST 2016 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:	Red Hat Enterprise Linux Workstation release 7.2 (Maipo)
```

**Comment by danlipsa** [comment]
I've seen the same error on Ubuntu 18.04

**Comment by adamjstewart** [comment]
Oops, sorry about that. That PR went through a lot of refactoring and it looks like that variable was lost in the process. I'll merge #12234 when it passes its tests.

**Comment by sethrj** [comment]
Yeah I figured as much -- rebasing has a tendency to do that. ;) No worries!

**Comment by adamjstewart** [comment]
Closed by #12234
